Makefile

```diff
@@ -1,29 +1,27 @@
 
-
 # define VERBOSE to see executed commands
-
+# default build configuration
 TARGET=hyped
-MAIN=main.cpp
+MAIN=run/main.cpp
+CROSS=0
+NOLINT=0
+
 SRCS_DIR:=src
 LIBS_DIR:=lib
 OBJS_DIR:=bin
 
-CFLAGS:=-pthread -std=c++11 -O2 -Wall -Wno-unused-result
-LFLAGS:=-lpthread -pthread
+include utils/config.mk
 
-# default configuration
-CROSS=0
-NOLINT=0
-PROTOBUF=0
+# libaries for generating TARGET
 EIGEN=$(LIBS_DIR)/Eigen
 
 ifeq ($(CROSS), 0)
-	CC:=g++
-	UNAME=$(shell uname)
 	ifneq ($(UNAME),Linux)
 		# assume Windows
 		UNAME='Windows'
-		CFLAGS:=$(CFLAGS) -DWIN
+		CFLAGS:=$(CFLAGS) -DWIN -std=gnu++11
```

Comment:
Interesting, what's the reason for using `gnu++11` here and `c++11` below? Just curious

Reply:
We found there were some compatibility issues when compiling GTest omn Cygwin using `-std=c++11`, switching to `gnu` seemed to fix it 😄